From 87b41094299cdc626876085549725ed0205ca830 Mon Sep 17 00:00:00 2001
From: Gilles Talis <gilles.talis@gmail.com>
Date: Sat, 26 Oct 2019 19:10:43 +0200
Subject: [PATCH] symbolize_unitests: some tests require HAVE_STACKTRACE
 defined to be run

The following test cases functions:
-  TestWithPCInsideInlineFunction
-  TestWithPCInsideNonInlineFunction
-  TestWithReturnAddress
are defined if HAVE_STACKTRACE is defined

Make sure they are invoked under the same condition

Signed-off-by: Gilles Talis <gilles.talis@gmail.com>
---
 src/symbolize_unittest.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/symbolize_unittest.cc b/src/symbolize_unittest.cc
index 9a17e7b..e11470e 100644
--- a/src/symbolize_unittest.cc
+++ b/src/symbolize_unittest.cc
@@ -407,9 +407,12 @@ int main(int argc, char **argv) {
   // used to install some callback function at InitGoogle() time.
   InstallSymbolizeCallback(NULL);

+# if defined(HAVE_STACKTRACE)
   TestWithPCInsideInlineFunction();
   TestWithPCInsideNonInlineFunction();
   TestWithReturnAddress();
+# endif
+
   return RUN_ALL_TESTS();
 # elif defined(OS_WINDOWS) || defined(OS_CYGWIN)
   TestWithReturnAddress();
-- 
2.17.1

